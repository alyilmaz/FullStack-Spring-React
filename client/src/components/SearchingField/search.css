.search-input {
  width: 200px;
  height: 45px;
  box-sizing: border-box;
  border: 2px solid #ccc;
  border-radius: 10px;
  font-size: 16px;
  background-color: white;
  background-image: url('../../../public/search.png');
  background-position: 10px 10px; 
  background-repeat: no-repeat;
  padding: 12px 20px 12px 40px;
  transition: width 0.4s ease-in-out;
  background-size: 20px;
}

.search-input:focus {
  width: 250px;
}