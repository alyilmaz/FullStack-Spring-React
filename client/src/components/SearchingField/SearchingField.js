import "./search.css"
const SearchingField = ({onChange, placeholder}) =>{

    return(
        <div className="search-container"> 
        <input
            className="search-input"
            key= "key"
            id="search"
            type="search"
            data-testid="search-name"
            placeholder=  {placeholder!== undefined ? placeholder : "searching"}
            onChange={event => onChange(event.target.value)}
        />
        </div>
    )
}

export default SearchingField;