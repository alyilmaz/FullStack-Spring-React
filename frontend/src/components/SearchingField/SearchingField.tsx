import "./search.css"

interface Props{
    placeholder?: string,
    onChange(e:React.ChangeEvent<HTMLInputElement>): void;
}


const SearchingField = (props:Props) =>{

    return(
        <div className="search-container"> 
        <input
            className="search-input"
            key= "key"
            id="search"
            type="search"
            placeholder=  {props.placeholder!== undefined ? props.placeholder : "searching"}
            onChange={event => props.onChange(event)}
        />
        </div>
    )
}

export default SearchingField;