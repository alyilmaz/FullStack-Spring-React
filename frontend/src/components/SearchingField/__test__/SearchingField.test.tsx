/* eslint-disable testing-library/no-node-access */
import SearchingField from "../SearchingField";
import {render, screen} from "@testing-library/react";
import renderer from 'react-test-renderer';

test("the searching field should be enabled", () =>{
    render(<SearchingField onChange={(e) => e.target.value}/>);
    expect(document.getElementById("search")).toBeEnabled();
});

test("the searching field get By its placeholder text", () =>{
    render(<SearchingField onChange={(e) => e.target.value}/>);
    expect(screen.getByPlaceholderText("searching")).toBeEnabled();
});

test("the searching field get By its placeholder via props", () =>{
    render(<SearchingField placeholder = "test" onChange={(e) => e.target.value}/>);
    expect(screen.getByPlaceholderText("test")).toBeEnabled();
});

test("create snapshot", () =>{
    const tree = renderer
                    .create(<SearchingField placeholder = "test" onChange={(e) => e.target.value}/>)
                    .toJSON();
    expect(tree).toMatchSnapshot();

})