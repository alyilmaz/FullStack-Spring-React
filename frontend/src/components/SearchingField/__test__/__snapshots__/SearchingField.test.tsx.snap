// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot 1`] = `
<div
  className="search-container"
>
  <input
    className="search-input"
    id="search"
    onChange={[Function]}
    placeholder="test"
    type="search"
  />
</div>
`;
